CustomPlugin/Environment/InvalidEnvConfigurationException.cs

```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace CustomPlugin.Environment
+{
+    public class InvalidEnvConfigurationException : Exception
+    {
+        public InvalidEnvConfigurationException(string message) : base(message)
+        {
```

Comment:
shalle we provide a customized error message upon the base message?

Reply:
I'm providing the message through the class that throws the Exception, for example `ConfigValidator.cs`. Do you think we should add a common message within the `InvalidConfigurationException` class?

Reply:
I'd expect the InvalidEnvConfigurationException to provide a simple base message: "Invalid configuration: .... " then any class can add it's own specific information.